src/main/java/seedu/recurrence/Recurrence.java

```diff
@@ -9,18 +9,22 @@
 
 import java.time.LocalDate;
 import java.time.YearMonth;
+import java.util.ArrayList;
 
 public abstract class Recurrence {
     protected int getLastDayOfMonth(LocalDate newEntry) {
         YearMonth date = YearMonth.of(newEntry.getYear(), newEntry.getMonth());
         return date.atEndOfMonth().getDayOfMonth();
     }
 
-    protected <T extends Type> void checkIfDateAltered(T newEntry, LocalDate checkDate) {
+    protected <T extends Type> void checkIfDateAltered(T newEntry, LocalDate checkDate, ArrayList<T> list) {
         int dayOfSupposedRecurrence = newEntry.getDayOfRecurrence();
         int lastDayOfNewEntryMonth = getLastDayOfMonth(newEntry.getDate());
         newEntry.editDateWithLocalDate(checkDate.withDayOfMonth(Math.min(dayOfSupposedRecurrence,
```

Comment:
Irrelevant to this PR, but this seems abit nested, may be better to separate it to a few lines or extract it out as a function

Reply:
can, will update when changed

Reply:
have resolved issue